jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -823,31 +435,52 @@ YUI.add('user-profile', function() {
       return (
         <div>
           {this._generateRows('models', state.envList, state.loadingModels)}
-          {this._generateRows(
-            'bundles', state.bundleList, state.loadingBundles)}
-          {this._generateRows('charms', state.charmList, state.loadingCharms)}
+          <juju.components.EntityList
+            changeState={this.props.changeState}
+            charmstore={this.props.charmstore}
+            getDiagramURL={this.props.getDiagramURL}
+            type='bundle'
+            user={this.props.user}
+            users={this.props.users} />
+          <juju.components.EntityList
+            changeState={this.props.changeState}
+            charmstore={this.props.charmstore}
+            getDiagramURL={this.props.getDiagramURL}
+            type='charm'
+            user={this.props.user}
+            users={this.props.users} />
           <juju.components.AgreementList
             getAgreements={this.props.getAgreements}
             user={this.props.user} />
-          {this._generateRows(
-            'budgets', state.budgetList, state.loadingBudgets)}
+          <juju.components.BudgetList
+            listBudgets={this.props.listBudgets}
+            user={this.props.user} />
         </div>);
     },
 
     render: function() {
       var username = this.props.user && this.props.user.usernameDisplay;
       var state = this.state;
+      /* XXX Find some way to percolate these up from the child components. */
+      /*
       var bundleCount = this._safeCount(state.bundleList);
       var charmCount = this._safeCount(state.charmList);
+      */
       var modelCount = this._safeCount(state.envList);
       var pluralize = this.props.pluralize;
+      /* XXX Should include agreements, budgets, etc. in these links. */
+      /*
       var links = [{
         label: `${modelCount} ${pluralize('model', modelCount)}`
       }, {
         label: `${bundleCount} ${pluralize('bundle', bundleCount)}`
       }, {
         label: `${charmCount} ${pluralize('charm', charmCount)}`
       }];
+      */
```

Comment:
Could you add some cards for these commented out sections (you may have, but I didn't see any at a quick glance)?